### PR TITLE
refactor: use runner-manager to all eg runners lifecycle

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

refactor: eg runners 

1. register and start runners:

image

2. shutdown runners:

image

**What this PR does / why we need it**:

use runner-manager to all eg runners lifecycle

**Which issue(s) this PR fixes**:

NONE

### CODE REVIEW

**Comment by arkodg** [comment]
hey @Xunzhuo sorry for the delay, will review, once rebased on the e2e pass

File: internal/globalratelimit/runner/runner.go [hunk 1]
```diff
@@ -44,93 +45,96 @@ const (
...
-	config.Server
+func Register(resources Resources, globalConfig config.Server) {
+	runner.Manager().Register(New(resources, globalConfig), runner.RootParentRunner)
+}
+
+func New(resources Resources, globalConfig config.Server) *ratelimitRunner {
+	return &ratelimitRunner{runner.New(string(v1alpha1.LogComponentGlobalRateLimitRunner), resources, globalConfig)}
+}
+
+type Resources struct {
```
Comment (arkodg) [inline internal/globalratelimit/runner/runner.go:56]: curious why you picked `Resources` over `Config` ?

**Comment by arkodg** [comment]
hey @Xunzhuo trying to better understand if this PR is needed for v0.5.0-rc1 or not, is this a refactor or does it also solve any bugs ?

**Comment by Xunzhuo** [comment]
Not for v0.5.0 I think

**Comment by Xunzhuo** [comment]
let me close this for now, since code structure has been changed a lot.

**Pull request closed without merging**